torch/_inductor/utils.py

```diff
@@ -795,3 +795,12 @@ def get_info_str(ms, n_regs, n_spills, shared, prefix=""):
         print(
             "No kernel with benchmark functionality found. Make sure you run inductor with config.benchmark_kernel being True"
         )
+
+
+@contextlib.contextmanager
+def maybe_profile(should_profile, *args, **kwargs):
```

Comment:
there's nearly the same function in benchmarks/dynamo/common.py, can you replace it there also?